From 0cdf90cc97113976f8b3b97d9ad24760f997b89d Mon Sep 17 00:00:00 2001
From: Matthew Gerlach <matthew.gerlach@intel.com>
Date: Fri, 9 Dec 2016 11:45:23 -0800
Subject: [PATCH 020/151] FogBugz #408870: add debugfs interface for fpga
 config complete timeout

commit  606e10bd4a04e77c74e8fc32007e68f7b012663b from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Add debugfs interface and documentation for setting the maximum
time in microseconds for the FPGA to go to the operating
state after the region has been programmed.

Signed-off-by: Matthew Gerlach <matthew.gerlach@linux.intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 Documentation/fpga/debugfs.txt  | 6 ++++++
 drivers/fpga/fpga-mgr-debugfs.c | 5 +++++
 2 files changed, 11 insertions(+)

diff --git a/Documentation/fpga/debugfs.txt b/Documentation/fpga/debugfs.txt
index 1b34d5460d5d..b01950f76e20 100644
--- a/Documentation/fpga/debugfs.txt
+++ b/Documentation/fpga/debugfs.txt
@@ -9,6 +9,12 @@ three files:
 
    $ echo 1 > /sys/kernel/debug/fpga_manager/fpga0/flags
 
+ - [RW] config_complete_timeout_us:	time out in microseconds to wait for
+					FPGA to go to operating state after
+					region has been programmed.
+
+   $ echo 4 > /sys/kernel/debug/fpga_manager/fpga0/config_complete_timeout_us
+
  - [RW] firmware_name:  Name of an FPGA image firmware file.  Writing initiates
                         a complete FPGA programming cycle.  Note that the image
                         file must be in a directory on the firmware search path
diff --git a/drivers/fpga/fpga-mgr-debugfs.c b/drivers/fpga/fpga-mgr-debugfs.c
index ae3c624f456a..941d87851e48 100644
--- a/drivers/fpga/fpga-mgr-debugfs.c
+++ b/drivers/fpga/fpga-mgr-debugfs.c
@@ -161,8 +161,13 @@ void fpga_mgr_debugfs_add(struct fpga_manager *mgr)
 			    &fpga_mgr_image_fops);
 
 	info = &debugfs->info;
+
 	debugfs_create_u32("flags", 0600, debugfs->debugfs_dir, &info->flags);
 
+	debugfs_create_u32("config_complete_timeout_us", 0600,
+			   debugfs->debugfs_dir,
+			   &info->config_complete_timeout_us);
+
 	mgr->debugfs = debugfs;
 }
 
-- 
2.26.1

